mailing_list/models.py

```diff
@@ -58,10 +65,32 @@ def get_or_create_or_delete_mailing_lists_for_canvas_course_id(self, canvas_cour
         canvas_sections = canvas_api_client.get_sections(canvas_course_id)
         mailing_lists_by_section_id = self._get_mailing_lists_by_section_id(canvas_course_id)
 
+        # get a list of the primary sections
+        primary_sections = [s['id'] for s in canvas_sections if s['sis_section_id'] is not None]
+
+        # if there is more than one primary section
+        # create a primary list by creating a primary section with an id of 'None'.
+        # This section will never be input into canvas, it's only for the purposes of
+        # the mailing list.
+        if len(primary_sections) > 1:
+            canvas_sections.append({
+                'integration_id': None,
+                'start_at': None,
+                'name': canvas_api_client.get_course(canvas_course_id)['course_code'],
+                'sis_import_id': None,
+                'end_at': None,
+                'sis_course_id': canvas_sections[0].get('sis_course_id'),
+                'sis_section_id': 'course',
+                'course_id': canvas_course_id,
+                'nonxlist_course_id': None,
+                'id': 'None'
+            })
```

Comment:
Do we need to create this fake section here at all? Why not just create the "all course" mailing list if it doesn't exist outside of the canvas_sections for loop below?